diff --git a/include/system/camera.h b/include/system/camera.h
index 5bc420a..c1806de 100644
--- a/include/system/camera.h
+++ b/include/system/camera.h
@@ -209,6 +209,8 @@ enum {
     CAMERA_CMD_HISTOGRAM_ON     = 11,
     CAMERA_CMD_HISTOGRAM_OFF     = 12,
     CAMERA_CMD_HISTOGRAM_SEND_DATA  = 13,
+    CAMERA_CMD_LONGSHOT_ON = 14,
+    CAMERA_CMD_LONGSHOT_OFF = 15,
     CAMERA_CMD_METADATA_ON = 100,
     CAMERA_CMD_METADATA_OFF = 101,
 
