diff --git a/config/common.mk b/config/common.mk
index aafd228..e4c75fc 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -45,7 +45,7 @@ $(eval TARGET_BOOTANIMATION_NAME := $(shell \
 endef
 $(foreach size,$(bootanimation_sizes), $(call check_and_set_bootanimation,$(size)))
 
-PRODUCT_BOOTANIMATION := vendor/cm/prebuilt/common/bootanimation/$(TARGET_BOOTANIMATION_NAME).zip
+#PRODUCT_BOOTANIMATION := vendor/cm/prebuilt/common/bootanimation/$(TARGET_BOOTANIMATION_NAME).zip
 endif
 
 ifdef CM_NIGHTLY
@@ -158,7 +158,7 @@ PRODUCT_PACKAGES += \
     Apollo \
     CMFileManager \
     LockClock \
-    CMAccount
+#    CMAccount
 
 # CM Hardware Abstraction Framework
 PRODUCT_PACKAGES += \
diff --git a/config/common_full.mk b/config/common_full.mk
index 428968a..98dfd57 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -8,7 +8,7 @@ include frameworks/base/data/sounds/NewAudio.mk
 include frameworks/base/data/sounds/AudioPackageNewWave.mk
 
 # Bring in all video files
-$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
+#$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
 
 # Include CM audio files
 include vendor/cm/config/cm_audio.mk
@@ -17,10 +17,10 @@ include vendor/cm/config/cm_audio.mk
 PRODUCT_PACKAGES += \
     Galaxy4 \
     HoloSpiralWallpaper \
-    LiveWallpapers \
-    LiveWallpapersPicker \
-    MagicSmokeWallpapers \
-    NoiseField \
+#    LiveWallpapers \
+#    LiveWallpapersPicker \
+#    MagicSmokeWallpapers \
+#    NoiseField \
     PhaseBeam \
     VisualizationWallpapers \
     PhotoTable
diff --git a/config/common_full_phone.mk b/config/common_full_phone.mk
index 2324a11..b2c0c5b 100644
--- a/config/common_full_phone.mk
+++ b/config/common_full_phone.mk
@@ -10,9 +10,9 @@ PRODUCT_PROPERTY_OVERRIDES += \
 PRODUCT_PACKAGES += \
   Mms
 
-ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
-    PRODUCT_COPY_FILES += \
-        vendor/cm/prebuilt/common/bootanimation/480.zip:system/media/bootanimation.zip
-endif
+#ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
+#    PRODUCT_COPY_FILES += \
+#        vendor/cm/prebuilt/common/bootanimation/480.zip:system/media/bootanimation.zip
+#endif
 
 $(call inherit-product, vendor/cm/config/telephony.mk)
diff --git a/config/common_full_tablet_wifionly.mk b/config/common_full_tablet_wifionly.mk
index 8bd6ee7..63de813 100644
--- a/config/common_full_tablet_wifionly.mk
+++ b/config/common_full_tablet_wifionly.mk
@@ -7,7 +7,7 @@ PRODUCT_PROPERTY_OVERRIDES += \
     ro.config.notification_sound=Argon.ogg \
     ro.config.alarm_alert=Hassium.ogg
 
-ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
-    PRODUCT_COPY_FILES += \
-        vendor/cm/prebuilt/common/bootanimation/800.zip:system/media/bootanimation.zip
-endif
+#ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
+#    PRODUCT_COPY_FILES += \
+#        vendor/cm/prebuilt/common/bootanimation/800.zip:system/media/bootanimation.zip
+#endif
diff --git a/config/common_mini_phone.mk b/config/common_mini_phone.mk
index 7589dad..eb296a7 100644
--- a/config/common_mini_phone.mk
+++ b/config/common_mini_phone.mk
@@ -16,9 +16,9 @@ PRODUCT_PROPERTY_OVERRIDES += \
 PRODUCT_PACKAGES += \
   Mms
 
-ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
-    PRODUCT_COPY_FILES += \
-        vendor/cm/prebuilt/common/bootanimation/320.zip:system/media/bootanimation.zip
-endif
+#ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
+#    PRODUCT_COPY_FILES += \
+#        vendor/cm/prebuilt/common/bootanimation/320.zip:system/media/bootanimation.zip
+#endif
 
 $(call inherit-product, vendor/cm/config/telephony.mk)
diff --git a/config/common_mini_tablet_wifionly.mk b/config/common_mini_tablet_wifionly.mk
index 9060001..c3c35e3 100644
--- a/config/common_mini_tablet_wifionly.mk
+++ b/config/common_mini_tablet_wifionly.mk
@@ -17,7 +17,7 @@ PRODUCT_PROPERTY_OVERRIDES += \
 PRODUCT_COPY_FILES += \
     system/bluetooth/data/main.nonsmartphone.conf:system/etc/bluetooth/main.conf
 
-ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
-    PRODUCT_COPY_FILES += \
-        vendor/cm/prebuilt/common/bootanimation/800.zip:system/media/bootanimation.zip
-endif
+#ifeq ($(TARGET_SCREEN_WIDTH) $(TARGET_SCREEN_HEIGHT),$(space))
+#    PRODUCT_COPY_FILES += \
+#        vendor/cm/prebuilt/common/bootanimation/800.zip:system/media/bootanimation.zip
+#endif
