commit f3c99068e3eafc999307c6ad8a61b6db2b8b6f58
Author: nx111 <gd.zhangdz@gmail.com>
Date:   Sun Aug 24 21:52:39 2014 +0800

    fix grallloc_priv.h, add display_config
    
    Change-Id: I019a6ad817152e9960b804049c0d7d6c33b10c3f

diff --git a/libgralloc/Android.mk b/libgralloc/Android.mk
index b8f69bf..aafe158 100644
--- a/libgralloc/Android.mk
+++ b/libgralloc/Android.mk
@@ -40,6 +40,7 @@ LOCAL_CFLAGS           := $(common_flags) -DLOG_TAG=\"memalloc\"
 LOCAL_SRC_FILES        := alloc_controller.cpp
 ifeq ($(TARGET_USES_ION),true)
     LOCAL_SRC_FILES +=  ionalloc.cpp
+    LOCAL_CFLAGS    +=  -DUSE_ION
 else
     LOCAL_SRC_FILES +=  pmemalloc.cpp \
                         ashmemalloc.cpp \
diff --git a/libgralloc/gralloc_priv.h b/libgralloc/gralloc_priv.h
index 8a3100f..fabaa04 100644
--- a/libgralloc/gralloc_priv.h
+++ b/libgralloc/gralloc_priv.h
@@ -78,6 +78,12 @@ enum {
      */
     GRALLOC_USAGE_PRIVATE_CP_BUFFER       =       0x00400000,
 
+    /* Buffer content should be displayed on an primary display only */
+    GRALLOC_USAGE_PRIVATE_INTERNAL_ONLY   =       0x00800000,
+
+    /* This flag is used for SECURE display usecase */
+    GRALLOC_USAGE_PRIVATE_SECURE_DISPLAY  =       0x10000000,
+
     /* Legacy heaps - these heaps are no-ops so we are making them zero
      * The flags need to be around to compile certain HALs which have
      * not cleaned up the code
