diff --git a/mac80211/wpa_supplicant_lib/driver_mac80211_nl.c b/mac80211/wpa_supplicant_lib/driver_mac80211_nl.c
index 5ae2a9b..891c63b 100644
--- a/mac80211/wpa_supplicant_lib/driver_mac80211_nl.c
+++ b/mac80211/wpa_supplicant_lib/driver_mac80211_nl.c
@@ -98,24 +98,3 @@ int wpa_driver_nl80211_driver_cmd(void *priv, char *cmd, char *buf,
 	return ret;
 }
 
-int wpa_driver_set_p2p_noa(void *priv, u8 count, int start, int duration)
-{
-	return -1;
-}
-
-int wpa_driver_get_p2p_noa(void *priv, u8 *buf, size_t len)
-{
-	return -1;
-}
-
-int wpa_driver_set_p2p_ps(void *priv, int legacy_ps, int opp_ps, int ctwindow)
-{
-	return -1;
-}
-
-int wpa_driver_set_ap_wps_p2p_ie(void *priv, const struct wpabuf *beacon,
-				 const struct wpabuf *proberesp,
-				 const struct wpabuf *assocresp)
-{
-	return 0;
-}
