diff --git a/services/batteryservice/BatteryProperties.cpp b/services/batteryservice/BatteryProperties.cpp
index 8c7d287..2aeb21a 100644
--- a/services/batteryservice/BatteryProperties.cpp
+++ b/services/batteryservice/BatteryProperties.cpp
@@ -37,7 +37,8 @@ status_t BatteryProperties::readFromParcel(Parcel* p) {
     batteryStatus = p->readInt32();
     batteryHealth = p->readInt32();
     batteryPresent = p->readInt32() == 1 ? true : false;
-    batteryLevel = p->readInt32();
+    int mbatteryLevel = p->readInt32();
+    batteryLevel = mbatteryLevel > 100 ? 100 : mbatteryLevel;
     batteryVoltage = p->readInt32();
     batteryCurrentNow = p->readInt32();
     batteryChargeCounter = p->readInt32();
@@ -49,7 +50,8 @@ status_t BatteryProperties::readFromParcel(Parcel* p) {
     dockBatteryStatus = p->readInt32();
     dockBatteryHealth = p->readInt32();
     dockBatteryPresent = p->readInt32() == 1 ? true : false;
-    dockBatteryLevel = p->readInt32();
+    int mdockBatteryLevel = p->readInt32();
+    dockBatteryLevel = mdockBatteryLevel > 100 ? 100 : mdockBatteryLevel;
     dockBatteryVoltage = p->readInt32();
     dockBatteryCurrentNow = p->readInt32();
     dockBatteryChargeCounter = p->readInt32();
