diff --git a/core/Makefile b/core/Makefile
index 6d0e8f3..012950b 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -873,7 +873,16 @@ ifdef is_tests_build
 #    $(tests_MODULES))
 endif
 
-FULL_SYSTEMIMAGE_DEPS := $(INTERNAL_SYSTEMIMAGE_FILES) $(INTERNAL_USERIMAGES_DEPS)
+## add squisher for clean some file before make systemimage
+systemimage-squisher: $(INTERNAL_SYSTEMIMAGE_FILES)
+ifeq ($(TARGET_SYSTEMIMAGE_USE_SQUISHER),true)
+	@echo -e ${CL_YLW}"Running squisher..."${CL_RST}
+	$(hide) APKCERTS=$(APKCERTS_FILE) ./vendor/cm/tools/squisher
+endif
+
+.PHONY: systemimage-squisher
+
+FULL_SYSTEMIMAGE_DEPS := $(INTERNAL_SYSTEMIMAGE_FILES) $(INTERNAL_USERIMAGES_DEPS) systemimage-squisher
 # -----------------------------------------------------------------
 # installed file list
 # Depending on anything that $(BUILT_SYSTEMIMAGE) depends on.
