diff --git a/core/java/android/hardware/Camera.java b/core/java/android/hardware/Camera.java
index 0764445..9fb9ce7 100644
--- a/core/java/android/hardware/Camera.java
+++ b/core/java/android/hardware/Camera.java
@@ -4746,7 +4746,7 @@ public class Camera {
          * @return max sharpness level
          */
          public int getMaxSharpness(){
-            return getInt(KEY_QC_MAX_SHARPNESS);
+            return getInt(KEY_QC_MAX_SHARPNESS,100);
          }
 
          /** @hide
@@ -4764,7 +4764,7 @@ public class Camera {
          * @return max contrast level
          */
          public int getMaxContrast(){
-            return getInt(KEY_QC_MAX_CONTRAST);
+            return getInt(KEY_QC_MAX_CONTRAST,100);
          }
 
          /** @hide
@@ -4782,7 +4782,7 @@ public class Camera {
          * @return max contrast level
          */
          public int getMaxSaturation(){
-            return getInt(KEY_QC_MAX_SATURATION);
+            return getInt(KEY_QC_MAX_SATURATION,100);
          }
 
          /** @hide
