diff --git a/core/jni/android_os_FileUtils.cpp b/core/jni/android_os_FileUtils.cpp
index b236480..53601e5 100644
--- a/core/jni/android_os_FileUtils.cpp
+++ b/core/jni/android_os_FileUtils.cpp
@@ -32,6 +32,7 @@
 #include <sys/ioctl.h>
 #include <linux/msdos_fs.h>
 #include <blkid/blkid.h>
+#include <uuid/uuid.h>
 
 namespace android {
 
@@ -74,6 +75,30 @@ jint android_os_FileUtils_getVolumeUUID(JNIEnv* env, jobject clazz, jstring path
     } else {
         uuid = blkid_get_tag_value(NULL, "UUID", pathStr);
     }
+
+    char uuidFilePath[256];
+    if (!uuid && findDevice && !strcmp(device,"/dev/fuse")){
+        FILE * fp;
+        snprintf(uuidFilePath,sizeof(uuidFilePath),"%s/.VolumeID",pathStr);
+        if (fp=fopen(uuidFilePath,"r")){
+            if(fscanf(fp,"%256s",line))
+               uuid=line;
+            fclose(fp);
+        }
+        else if(fp=fopen(uuidFilePath,"w")){
+             uuid_t uu;
+             uuid_generate(uu);
+             int i;
+             for(i=0;i<sizeof(uuid_t);i++){
+		sprintf(line+2*i,"%02x",uu[i]);
+	     }
+             line[2*i]='\0';
+             uuid=line;
+             fprintf(fp,"%s",uuid);
+             fclose(fp);
+        }
+    }
+
     if (uuid) {
         ALOGD("UUID for %s is %s\n", pathStr, uuid);
 
