diff --git a/drivers/misc/omap_temp_sensor.c b/drivers/misc/omap_temp_sensor.c
index 8ecb1bf..090765f 100644
--- a/drivers/misc/omap_temp_sensor.c
+++ b/drivers/misc/omap_temp_sensor.c
@@ -420,7 +420,11 @@ static void throttle_delayed_work_fn(struct work_struct *work)
 					     throttle_work.work);
 	curr = omap_read_current_temp(temp_sensor);
 
+#ifdef CONFIG_OMAP_TEMP_CONTROL
 	if (curr >= hot_threshold || curr < 0) {
+#else
+	if (curr >=BGAP_THRESHOLD_T_HOT || curr < 0) {
+#endif
 		pr_warn("%s: OMAP temp read %d exceeds the threshold\n",
 			__func__, curr);
 		omap_thermal_throttle();
@@ -492,7 +496,9 @@ static int __devinit omap_temp_sensor_probe(struct platform_device *pdev)
 	}
 
 	temp_sensor = kzalloc(sizeof(struct omap_temp_sensor), GFP_KERNEL);
+#ifdef CONFIG_OMAP_TEMP_CONTROL
 	ctrl_sensor = temp_sensor;
+#endif
 	
 	if (!temp_sensor)
 		return -ENOMEM;
