diff --git a/core/jni/android_view_SurfaceControl.cpp b/core/jni/android_view_SurfaceControl.cpp
index 67eade8..2302327 100644
--- a/core/jni/android_view_SurfaceControl.cpp
+++ b/core/jni/android_view_SurfaceControl.cpp
@@ -163,10 +163,12 @@ static inline SkBitmap::Config convertPixelFormat(PixelFormat format) {
 static jobject nativeScreenshotBitmap(JNIEnv* env, jclass clazz, jobject displayTokenObj,
         jint width, jint height, jint minLayer, jint maxLayer, bool allLayers) {
     sp<IBinder> displayToken = ibinderForJavaObject(env, displayTokenObj);
+    ALOGD("nativeScreenshotBitmap 0");
     if (displayToken == NULL) {
         return NULL;
     }
-
+    
+    ALOGD("nativeScreenshotBitmap 1");
     ScreenshotPixelRef* pixels = new ScreenshotPixelRef(NULL);
     if (pixels->update(displayToken, width, height,
             minLayer, maxLayer, allLayers) != NO_ERROR) {
@@ -194,7 +196,9 @@ static jobject nativeScreenshotBitmap(JNIEnv* env, jclass clazz, jobject display
         delete pixels;
         bitmap->setPixels(NULL);
     }
-
+    
+    ALOGD("nativeScreenshotBitmap 2");
+    
     return GraphicsJNI::createBitmap(env, bitmap,
             GraphicsJNI::kBitmapCreateFlag_Premultiplied, NULL);
 }
