diff --git a/services/surfaceflinger/Android.mk b/services/surfaceflinger/Android.mk
index 7b7a49b..9a86e02 100644
--- a/services/surfaceflinger/Android.mk
+++ b/services/surfaceflinger/Android.mk
@@ -76,6 +76,10 @@ else
     LOCAL_CFLAGS += -DPRESENT_TIME_OFFSET_FROM_VSYNC_NS=0
 endif
 
+ifeq ($(TARGET_USES_OPENGLES_FOR_SCREEN_CAPTURE),true)
+       LOCAL_CFLAGS += -DUSE_OPENGLES_FOR_SCREEN_CAPTURE
+endif
+
 LOCAL_CFLAGS += -fvisibility=hidden
 
 LOCAL_SHARED_LIBRARIES := \
diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 529461e..6a66584 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -3133,9 +3133,13 @@ status_t SurfaceFlinger::captureScreenImplLocked(
     status_t result = NO_ERROR;
     if (native_window_api_connect(window, NATIVE_WINDOW_API_EGL) == NO_ERROR) {
         uint32_t usage = GRALLOC_USAGE_SW_READ_OFTEN | GRALLOC_USAGE_SW_WRITE_OFTEN;
+#ifdef USE_OPENGLES_FOR_SCREEN_CAPTURE
+            usage |= GRALLOC_USAGE_HW_RENDER | GRALLOC_USAGE_HW_TEXTURE;
+#else
         if (!useReadPixels) {
             usage |= GRALLOC_USAGE_HW_RENDER | GRALLOC_USAGE_HW_TEXTURE;
         }
+#endif
 
         int err = 0;
         err = native_window_set_buffers_dimensions(window, reqWidth, reqHeight);
