diff --git a/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java b/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java
index 2a4e4f6..a854919 100644
--- a/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java
+++ b/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java
@@ -464,21 +464,23 @@ class GlobalScreenshot {
             return;
         }
 
+	Bitmap ss = Bitmap.createBitmap(mDisplayMetrics.widthPixels,
+		mDisplayMetrics.heightPixels, Bitmap.Config.ARGB_8888);
+	Canvas c = new Canvas(ss);
+
         if (requiresRotation) {
             // Rotate the screenshot to the current orientation
-            Bitmap ss = Bitmap.createBitmap(mDisplayMetrics.widthPixels,
-                    mDisplayMetrics.heightPixels, Bitmap.Config.ARGB_8888);
-            Canvas c = new Canvas(ss);
             c.translate(ss.getWidth() / 2, ss.getHeight() / 2);
             c.rotate(degrees);
             c.translate(-dims[0] / 2, -dims[1] / 2);
-            c.drawBitmap(mScreenBitmap, 0, 0, null);
-            c.setBitmap(null);
-            // Recycle the previous bitmap
-            mScreenBitmap.recycle();
-            mScreenBitmap = ss;
         }
 
+        c.drawBitmap(mScreenBitmap, 0, 0, null);
+        c.setBitmap(null);
+        // Recycle the previous bitmap
+        mScreenBitmap.recycle();
+        mScreenBitmap = ss;
+
         // Optimizations
         mScreenBitmap.setHasAlpha(false);
         mScreenBitmap.prepareToDraw();
