diff --git a/config/common.mk b/config/common.mk
index 1653d4b..eebaee3 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -30,9 +30,7 @@ endef
 $(foreach size,$(bootanimation_sizes), $(call check_and_set_bootanimation,$(size)))
 
 ifeq ($(TARGET_BOOTANIMATION_HALF_RES),true)
-PRODUCT_BOOTANIMATION := vendor/cm/prebuilt/common/bootanimation/halfres/$(TARGET_BOOTANIMATION_NAME).zip
 else
-PRODUCT_BOOTANIMATION := vendor/cm/prebuilt/common/bootanimation/$(TARGET_BOOTANIMATION_NAME).zip
 endif
 endif
 
@@ -138,7 +136,7 @@ PRODUCT_PACKAGES += \
 
 # Custom CM packages
 PRODUCT_PACKAGES += \
-    Launcher3 \
+#    Launcher3 \
     Trebuchet \
     DSPManager \
     libcyanogen-dsp \
@@ -148,8 +146,6 @@ PRODUCT_PACKAGES += \
     CMFileManager \
     LockClock \
     CMUpdater \
-    CMFota \
-    CMAccount
 
 # CM Hardware Abstraction Framework
 PRODUCT_PACKAGES += \
@@ -244,7 +240,7 @@ ifdef CM_BUILDTYPE
     ifneq ($(CM_BUILDTYPE), SNAPSHOT)
         ifdef CM_EXTRAVERSION
             # Force build type to EXPERIMENTAL
-            CM_BUILDTYPE := EXPERIMENTAL
+#            CM_BUILDTYPE := EXPERIMENTAL
             # Remove leading dash from CM_EXTRAVERSION
             CM_EXTRAVERSION := $(shell echo $(CM_EXTRAVERSION) | sed 's/-//')
             # Add leading dash to CM_EXTRAVERSION
@@ -253,7 +249,7 @@ ifdef CM_BUILDTYPE
     else
         ifndef CM_EXTRAVERSION
             # Force build type to EXPERIMENTAL, SNAPSHOT mandates a tag
-            CM_BUILDTYPE := EXPERIMENTAL
+#            CM_BUILDTYPE := EXPERIMENTAL
         else
             # Remove leading dash from CM_EXTRAVERSION
             CM_EXTRAVERSION := $(shell echo $(CM_EXTRAVERSION) | sed 's/-//')
diff --git a/config/common_full.mk b/config/common_full.mk
index 2eccc56..f584fe3 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -14,10 +14,6 @@ PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
 PRODUCT_PACKAGES += \
     Galaxy4 \
     HoloSpiralWallpaper \
-    LiveWallpapers \
-    LiveWallpapersPicker \
-    MagicSmokeWallpapers \
-    NoiseField \
     PhaseBeam \
     VisualizationWallpapers \
     PhotoTable \
