diff --git a/arch/arm/mach-omap2/board-mapphone.c b/arch/arm/mach-omap2/board-mapphone.c
index 48bb941..ebde3a9 100644
--- a/arch/arm/mach-omap2/board-mapphone.c
+++ b/arch/arm/mach-omap2/board-mapphone.c
@@ -174,7 +174,6 @@ static int plat_uart_disable(void)
 		if (!err)
 			uart_req = false;
 	}
-	wake_unlock(&st_wk_lock);
 	return err;
 }
 
@@ -189,6 +188,23 @@ static int plat_uart_enable(void)
 		if (!err)
 			uart_req = true;
 	}
+	return err;
+}
+
+static int plat_uart_asleep(void)
+{
+	int err;
+
+	err = plat_uart_disable();
+	wake_unlock(&st_wk_lock);
+	return err;
+}
+
+static int plat_uart_awake(void)
+{
+	int err;
+
+	err = plat_uart_enable();
 	wake_lock(&st_wk_lock);
 	return err;
 }
@@ -240,8 +256,8 @@ static struct ti_st_plat_data wilink_pdata = {
 	.baud_rate = 3686400,
 	.suspend = plat_wlink_kim_suspend,
 	.resume = plat_wlink_kim_resume,
-	.chip_asleep = plat_uart_disable,
-	.chip_awake  = plat_uart_enable,
+	.chip_asleep = plat_uart_asleep,
+	.chip_awake  = plat_uart_awake,
 	.chip_enable = plat_uart_enable,
 	.chip_disable = plat_uart_disable,
 };
